rdmo/management/assets/js/actions/elementActions.js

```diff
@@ -0,0 +1,632 @@
+      })
+      .catch(error => dispatch(storeElementError(element, error)))
+  }
+}
+
+export function storeElementInit(element) {
+  return {type: 'elements/storeElementInit', element}
+}
+
+export function storeElementSuccess(element) {
+  return {type: 'elements/storeElementSuccess', element}
+}
+
+export function storeElementError(element, error) {
+  return {type: 'elements/storeElementError', element, error}
+}
+
+// createElement
+
+export function createElement(elementType, parent={}) {
+  return function(dispatch, getState) {
+    updateLocation(getState().config.baseUrl, elementType, null, 'create')
+
+    dispatch(createElementInit(elementType))
+
+    let action
+    switch (elementType) {
+      case 'catalogs':
+        action = () => Promise.all([
+          QuestionsFactory.createCatalog(getState().config),
+          QuestionsApi.fetchSections('index')
+        ]).then(([element, sections]) => ({
+          element, sections
+        }))
+        break
+
+      case 'sections':
+        action = () => Promise.all([
+          QuestionsFactory.createSection(getState().config, parent),
+          QuestionsApi.fetchPages('index'),
+        ]).then(([element, pages]) => ({
+          element, parent, pages
+        }))
+        break
+
+      case 'pages':
+        action = () => Promise.all([
+          QuestionsFactory.createPage(getState().config, parent),
+          DomainApi.fetchAttributes('index'),
+          ConditionsApi.fetchConditions('index'),
+          QuestionsApi.fetchQuestionSets('index'),
+          QuestionsApi.fetchQuestions('index')
+        ]).then(([element, attributes, conditions,
+                  questionsets, questions]) => ({
+          element, parent, attributes, conditions, questionsets, questions
+        }))
+        break
+
+      case 'questionsets':
+        action = () => Promise.all([
+          QuestionsFactory.createQuestionSet(getState().config, parent),
+          DomainApi.fetchAttributes('index'),
+          ConditionsApi.fetchConditions('index'),
+          QuestionsApi.fetchQuestionSets('index'),
+          QuestionsApi.fetchQuestions('index')
+        ]).then(([element, attributes, conditions,
+                  questionsets, questions]) => ({
+          element, parent, attributes, conditions, questionsets, questions
+        }))
+        break
+
+      case 'questions':
+        action = () => Promise.all([
+          QuestionsFactory.createQuestion(getState().config, parent),
+          DomainApi.fetchAttributes('index'),
+          OptionsApi.fetchOptionSets('index'),
+          OptionsApi.fetchOptions('index'),
+          ConditionsApi.fetchConditions('index'),
+          QuestionsApi.fetchWidgetTypes(),
+          QuestionsApi.fetchValueTypes()
+        ]).then(([element, attributes, optionsets,
+                  options, conditions]) => ({
+            element, parent, attributes, optionsets, options, conditions
+          }))
+        break
+
+      case 'attributes':
```

Comment:
For a user flow: conditions > new > create new attribute > create > User can not create a new attribute for an "uncreated" condition. The user is stuck and can only go back.  
> This attribute will be added to the condition `undefined`.

Should the "create new attribute" button be blocked in this case or is it bug somehow? To prevent a user getting stuck there..

Reply:
This should now be fixed, the button is disabled as long as the new element is not saved.